crates/client/rpc/src/lib.rs

```diff
@@ -1694,39 +1815,58 @@ where
+    timestamp_in_millisecond / 1000
+}
 
-    Ok(extrinsic)
+fn is_pending_block(block_id: BlockId) -> bool {
+    block_id == BlockId::Tag(BlockTag::Pending)
 }
 
-fn convert_error<C, B, T>(
-    client: Arc<C>,
-    best_block_hash: <B as BlockT>::Hash,
-    call_result: Result<T, DispatchError>,
-) -> Result<T, StarknetRpcApiError>
-where
-    B: BlockT,
-    C: ProvideRuntimeApi<B>,
-    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
-{
-    match call_result {
-        Ok(val) => Ok(val),
-        Err(e) => match client.runtime_api().convert_error(best_block_hash, e) {
-            Ok(starknet_error) => Err(starknet_error.into()),
-            Err(_) => Err(StarknetRpcApiError::InternalServerError),
-        },
+fn starknet_api_to_starknet_core_event(event: starknet_api::transaction::Event) -> starknet_core::types::Event {
+    starknet_core::types::Event {
+        from_address: Felt252Wrapper::from(event.from_address).0,
+        keys: event.content.keys.into_iter().map(|felt| Felt252Wrapper::from(felt).0).collect(),
+        data: event.content.data.0.into_iter().map(|felt| Felt252Wrapper::from(felt).0).collect(),
+    }
+}
+
+fn starknet_api_to_starknet_core_message_to_l1(
+    message: starknet_api::transaction::MessageToL1,
+) -> starknet_core::types::MsgToL1 {
+    let mut to_address = [0u8; 32];
+    to_address[12..32].copy_from_slice(message.to_address.0.as_bytes());
+    starknet_core::types::MsgToL1 {
+        from_address: Felt252Wrapper::from(message.from_address).0,
+        to_address: FieldElement::from_bytes_be(&to_address).unwrap(),
+        payload: message.payload.0.into_iter().map(|felt| Felt252Wrapper::from(felt).0).collect(),
+    }
+}
+
+fn revert_error_to_execution_result(revert_error: Option<String>) -> ExecutionResult {
+    match revert_error {
+        None => ExecutionResult::Succeeded,
+        Some(message) => ExecutionResult::Reverted { reason: message },
+    }
+}
+
+fn actual_resources_to_execution_resources(resources: ResourcesMapping) -> ExecutionResources {
+    let resources = resources.0.into_iter().map(|(k, v)| (k.to_lowercase(), v)).collect::<HashMap<String, u64>>();
+    // Based on `VM_RESOURCE_FEE_COSTS`
+    // in crates/primitives/fee/src/lib.rs
+    ExecutionResources {
+        steps: *resources.get("n_steps").unwrap_or(&0),
+        // Missing param
+        memory_holes: None,
+        range_check_builtin_applications: *resources.get("range_check_builtin").unwrap_or(&0),
+        pedersen_builtin_applications: *resources.get("pedersen_builtin").unwrap_or(&0),
+        poseidon_builtin_applications: *resources.get("poseidon_builtin").unwrap_or(&0),
+        ec_op_builtin_applications: *resources.get("ec_op_builtin").unwrap_or(&0),
+        ecdsa_builtin_applications: *resources.get("ecdsa_builtin").unwrap_or(&0),
+        bitwise_builtin_applications: *resources.get("bitwise_builtin").unwrap_or(&0),
+        // Missing param
+        keccak_builtin_applications: 0,
```

Comment:
How is it missing? Can't you do:
```rust
bitwise_builtin_applications: *resources.get("keccak_builtin").unwrap_or(&0),
```
?

Reply:
I can but I see no point in doing this in that way because this param is missing here:
```rust
pub static VM_RESOURCE_FEE_COSTS: [(&str, FixedU128); 7] = [
    ("n_steps", FixedU128::from_inner(10_000_000_000_000_000)),
    ("pedersen_builtin", FixedU128::from_inner(320_000_000_000_000_000)),
    ("range_check_builtin", FixedU128::from_inner(160_000_000_000_000_000)),
    ("ecdsa_builtin", FixedU128::from_inner(20_480_000_000_000_000_000)),
    ("bitwise_builtin", FixedU128::from_inner(640_000_000_000_000_000)),
    ("poseidon_builtin", FixedU128::from_inner(320_000_000_000_000_000)),
    ("ec_op_builtin", FixedU128::from_inner(10_240_000_000_000_000_000)),
];

```

Reply:
In the future we will use the blockifier impl itself. And if they decide to add it later on and it already contains the keccak_builtin.
Right now the RPC lib we use is in the future in comparison to the blockifier one. But it will be fixed later and adding it will make the update frictionless